src/rshapes.c

```diff
@@ -1734,6 +1736,23 @@ bool CheckCollisionPointLine(Vector2 point, Vector2 p1, Vector2 p2, int threshol
     return collision;
 }
 
+bool CheckCollisionRayLine(Vector2 rayPosition, Vector2 rayDirection, Vector2 startPos, Vector2 endPos, Vector2 *collisionPoint)
+{    
+    bool collision = false;
+    Vector2 offsetPosition;
+    offsetPosition.x = rayPosition.x + rayDirection.x;
+    offsetPosition.y = rayPosition.y + rayDirection.y;
+
+    float denom = (startPos.x - endPos.x)*(rayPosition.y-offsetPosition.y) - (startPos.y - endPos.y)*(rayPosition.x-offsetPosition.x);
+    if (denom == 0) return collision;
+    float t = ((startPos.x - rayPosition.x)*(rayPosition.y - offsetPosition.y) - (startPos.y-rayPosition.y)*(rayPosition.x-offsetPosition.x)) / denom;
+    float u = ((startPos.x - rayPosition.x)*(startPos.y - endPos.y) - (startPos.y-rayPosition.y)*(startPos.x - endPos.x)) / denom;
+    collision = (t >= 0 && t <= 1 && u >= 0);
+    if (collision)
+        *collisionPoint = (Vector2){startPos.x + t * (endPos.x-startPos.x),startPos.y + t * (endPos.y-startPos.y)};
```

Comment:
You need to verify that the passed in collisionPoint is valid also. If NULL is passed in this will crash.
Making a temporary vector2 is also not necessary, you can just directly assign into the pointer's fields. See how CheckCollisionLines does it. This function should be nearly identical to that one.

Reply:
I completely forgot about that, thanks!